Marlin/gcode.h

```diff
@@ -148,6 +169,9 @@ class GCodeParser {
   // Code value pointer was set
   FORCE_INLINE static bool has_value() { return value_ptr != NULL; }
 
+  // Seen and has value
+  FORCE_INLINE static bool seenval(const char c) { return seen(c) && has_value(); }
```

Comment:
Yes, of course.  There are places it makes sense, and places where things should be handled better/differently...